pkg/apis/gke.cattle.io/v1/types.go

```diff
@@ -0,0 +1,86 @@
+
+type GKEClusterConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   GKEClusterConfigSpec   `json:"spec"`
+	Status GKEClusterConfigStatus `json:"status"`
+}
+
+// GKEClusterConfigSpec is the spec for a GKEClusterConfig resource
+type GKEClusterConfigSpec struct {
+	DisplayName        string            `json:"displayName" norman:"noupdate"`
+	Region             string            `json:"region" norman:"noupdate"`
+	Imported           bool              `json:"imported" norman:"noupdate"`
+	Description        string            `json:"description"`
+	EnableAlphaFeature bool              `json:"enableAlphaFeature"`
+	ClusterAddons      ClusterAddons     `json:"clusterAddons"`
+	ClusterIpv4Cidr    string            `json:"clusterIpv4Cidr"`
+	ProjectID          string            `json:"projectID"`
+	CredentialContent  string            `json:"credentialContent"`
+	ClusterName        string            `json:"clusterName"`
+	KubernetesVersion  *string           `json:"kubernetesVersion"`
+	Tags               map[string]string `json:"tags"`
+	SecretsEncryption  *bool             `json:"secretsEncryption" norman:"noupdate"`
+	LoggingTypes       []string          `json:"loggingTypes"`
+	Subnets            []string          `json:"subnets" norman:"noupdate"`
+	SecurityGroups     []string          `json:"securityGroups" norman:"noupdate"`
+	ServiceRole        *string           `json:"serviceRole" norman:"noupdate"`
+	NodeGroups         []NodeGroup       `json:"nodeGroups"`
+}
+
+type GKEClusterConfigStatus struct {
+	Phase          string   `json:"phase"`
+	VirtualNetwork string   `json:"virtualNetwork"`
+	Subnets        []string `json:"subnets"`
+	SecurityGroups []string `json:"securityGroups"`
+	// describes how the above network fields were provided. Valid values are provided and generated
+	NetworkFieldsSource string `json:"networkFieldsSource"`
+	FailureMessage      string `json:"failureMessage"`
+}
+
+type ClusterAddons struct {
+	HTTPLoadBalancing        *bool `json:"httpLoadBalancing`
+	HorizontalPodAutoscaling *bool `json:"horizontalPodAutoscaling"`
+	KubernetesDashboard      *bool `json:"kubernetesDashboard"`
+	NetworkPolicyConfig      *bool `json:"networkPolicyConfig"`
+}
+
+type NodeGroup struct {
+	Gpu           *bool              `json:"gpu"`
+	NodegroupName *string            `json:"nodegroupName" norman:"required"`
+	DiskSize      *int64             `json:"diskSize"`
+	InstanceType  *string            `json:"instanceType" norman:"required"`
+	Labels        map[string]*string `json:"labels"`
+	Ec2SshKey     *string            `json:"ec2SshKey"`
+	DesiredSize   *int64             `json:"desiredSize"`
+	MaxSize       *int64             `json:"maxSize"`
+	MinSize       *int64             `json:"minSize"`
+	Subnets       []string           `json:"subnets"`
+	Tags          map[string]*string `json:"tags"`
+	Version       *string            `json:"version"`
+}
```

Comment:
Remove, unless this is somehow not EKS specific.

Reply:
well. NodePools are a GKE thing. in my newest commit this has been updated to reflect the right name and updated fields. I will include the newest commit in this PR as it seems to solve a lot of "this is a skeleton app" problems you all are pointing out. This initial PR was just to get something out there you all could see :+1: Thanks!